graylog2-server/src/main/java/org/graylog2/users/UserImpl.java

```diff
@@ -226,7 +227,7 @@ public void setEmail(final String email) {
 
     @Override
     public List<String> getPermissions() {
-        final Set<String> permissionSet = new HashSet<>(this.permissions.userSelfEditPermissions(getName()));
+        final Set<String> permissionSet = isServiceAccount() ? new HashSet<>() : new HashSet<>(this.permissions.userSelfEditPermissions(getName()));
```

Comment:
Please check all usages of `Permissions#userSelfEditPermissions()` if we need to add more `isServiceAccount` checks. There is at least one usage in `PermissionsResource` where I am not sure if we need to check for a service account or not.

Reply:
`PermissionsResource` and `Permissions` both have a `readerPermissions()` method that call `userSelfEditPermissions()`. Currently neither class accesses the DB - they operate purely on permissions. The user name is just used to build permission strings. 
`Permissions#readerPermissions` is only used in unit tests. 
`PermissionsResource#readerPermissions` is not currently used in the frontend.

I think it's reasonable to keep these classes agnostic of service accounts.

Reply:
Alright, sounds good. Thanks for the investigation. :+1: